src/hello-from-rust.md

```diff
@@ -0,0 +1,186 @@
+   Compiling core v0.0.0 (file:///home/steve/src/intermezzOS/kernel/chapter_05/target/libcore)
+RUSTFLAGS="-L target/libcore/target/x86_64-unknown-intermezzos-gnu/release" cargo build --target x86_64-unknown-intermezzos-gnu.json
+   Compiling intermezzos v0.1.0 (file:///home/steve/src/intermezzOS/kernel/chapter_05)
+$
+```
+
+Success! It should all build properly. There's one more thing I'd like to note
+about this makefile: in a strict sense, it will try and rebuild too much. But
+watch what happens if we try to build a second time:
+
+```bash
+$ make target/x86_64-unknown-intermezzos-gnu/release/libintermezzos.a
+cp x86_64-unknown-intermezzos-gnu.json target/libcore
+cd target/libcore && cargo build --release --features disable_float --target=x86_64-unknown-intermezzos-gnu.json
+RUSTFLAGS="-L target/libcore/target/x86_64-unknown-intermezzos-gnu/release" cargo build --target x86_64-unknown-intermezzos-gnu.json
+$
+```
+
+We issued some commands, but didn't actually compile anything. With this
+layout, we're letting Cargo worry if stuff needs to be rebuilt. This makes
+our Makefile a bit easier to write, and also a bit more reliable. Cargo
+knows what it needs to do, let's just trust it to do the right thing.
+
+Now that we have it building, we need to modify the rule that builds the kernel
+to include `libintermezzos.a`:
+
+```make
+target/kernel.bin: target/multiboot_header.o target/boot.o src/asm/linker.ld target/x86_64-unknown-intermezzos-gnu/release/libintermezzos.a
+        ld -n -o target/kernel.bin -T src/asm/linker.ld target/multiboot_header.o target/boot.o target/x86_64-unknown-intermezzos-gnu/release/libintermezzos.a
+```
+
+And then we can build:
+
+```bash
+$ make
+mkdir -p target
+nasm -f elf64 src/asm/multiboot_header.asm -o target/multiboot_header.o
+mkdir -p target
+nasm -f elf64 src/asm/boot.asm -o target/boot.o
+cp x86_64-unknown-intermezzos-gnu.json target/libcore
+cd target/libcore && cargo build --release --features disable_float --target=x86_64-unknown-intermezzos-gnu.json
+RUSTFLAGS="-L target/libcore/target/x86_64-unknown-intermezzos-gnu/release" cargo build --release --target x86_64-unknown-intermezzos-gnu.json
+ld -n -o target/kernel.bin -T src/asm/linker.ld target/multiboot_header.o target/boot.o target/x86_64-unknown-intermezzos-gnu/release/libintermezzos.a
+$
+```
+
+Hooray! We are now successfully building our assembly code and our Rust code, and then putting them together.
+
+Now, to write our Rust. Add this function to `src/lib.rs`:
+
+```rust
+#[no_mangle]
+pub extern fn kmain() -> ! {
+
+    loop { }
+}
+```
+
+This is our main function, which is traditionally called `kmain()`, for 'kernel
+main.' We need to use the `#[no_mangle]` and `pub extern` annotations to indicate
+that we're going to call this function like we would call a C function. The `-> !`
+indicates that this function never returns. And in fact, it does not: the body
```

Comment:
Hmmm, I think this is one of things things in which English is ambiguous. 

> And in fact, it does not return

is how I read this, but you're reading it as negating the "never returns."